crates/agent/src/directives/illegal_tenant_names.rs

```diff
@@ -0,0 +1,1481 @@
+use phf::phf_map;
+
+pub const ILLEGAL_TENANT_NAMES: phf::Map<&str, &str> = phf_map! {
```

Comment:
We need this to be managed in the DB, rather than in source code, as we need to be able to update this list and/or react quickly without having to do a code push / build / release.

We had talked about just creating sentinel entries in the tenants table with zero quota which prevent others from creating a tenant of the same name (case invariant). Another option would be to create this as a table in the `internal` schema and update our logic to check it.

Reply:
I think I'll push a little more strongly for a separate "block list" table in the `internal` schema of the DB.
While we _could_ put these it into `tenants` directly, I'm just thinking about the potential for confusion with future operations people who might not _entirely_ know what they're doing.